index.js

```diff
@@ -154,28 +153,52 @@ ExtractTextPlugin.prototype.applyAdditionalInformation = function(source, info)
 };
 
 ExtractTextPlugin.prototype.loader = function(options) {
-	options = JSON.parse(JSON.stringify(options || {}));
-	options.id = this.id;
-	return ExtractTextPlugin.loader(options);
+	return ExtractTextPlugin.loader(mergeOptions({id: this.id}, options));
 };
 
-ExtractTextPlugin.prototype.extract = function(before, loader, options) {
-	if(typeof loader === "string" || Array.isArray(loader)) {
-		if(typeof before === "string") {
-			before = before.split("!");
-		}
-		return [
-			this.loader(mergeOptions({omit: before.length, extract: true, remove: true}, options))
-		].concat(before, loader).join("!");
-	} else {
-		options = loader;
-		loader = before;
-		return [
-			this.loader(mergeOptions({remove: true}, options))
-		].concat(loader).join("!");
+ExtractTextPlugin.prototype.extractAll = function(options) {
+	if(arguments.length > 1) {
+		throw new Error("Deprecation notice: extract now only takes a single argument. Either an options " +
```

Comment:
Good point about <q>breaking change</q> I should add a migration note to the README as well.